### DISCUSSION TITLE
Program stuck running after crash

### DISCUSSION THREAD

**Discussion by SirZeeno** [AvaloniaUI/Avalonia#15337] [Q&A]
### Describe the bug

I am not entirely sure if this just has to do with the way Avalonia crashes or if its the Rider IDE, but everytime it crashes because the carouselItems arent loaded in yet and i am trying to use the Next function on the carouselItems, it leaves the process running in the background that i have to go into the processes in task manager to close the program completely before i can recompile and test a new solution to bugs.

### To Reproduce

make the program crash with error code like mine after launching the program through the run button in the editor. (error code for reference in picture)
![image](https://github.com/AvaloniaUI/Avalonia/assets/79223742/436bac52-de08-43b5-bfcc-2fc46d329ce6)

### Expected behavior

the program closes and doesnt leave the dll or exe running in system processes

### Avalonia version

11.0.10

### OS

Windows

**Comment by SirZeeno** [comment]
Here is my Project repo, disregard that its a mess rn, i am trying to merge the CLI version of my program into this and i havent gotten far yet.

https://github.com/SirZeeno/Anime-Archive-Handler-GUI

**Comment by maxkatz6** [comment]
Well, it crashes in this your method:
https://github.com/SirZeeno/Anime-Archive-Handler-GUI/blob/master/Anime%20Archive%20Handler%20GUI/Anime%20Archive%20Handler%20GUI/Views/MainView.axaml.cs#L307-L315

Not sure if  `var Slides = ContentControl.FindControl<Carousel>("Slides");` is supposed to work like that.

**Reply by SirZeeno** [reply]
Sorry for the Repost from what i already commented, I just wasn't sure if it was being seen.

Honestly im not sure either, i am pretty new to Avalonia and am mostly doing a lot of testing rn to see what works and what i need to do in certain situations like this where i need to reference a name in a DataTemplate. But my point on the issue is not what causes the crash, but the fact the process stays running after the crash and is only visible when going in the details tab in the task manager. it basically locks up the entire application from starting or being compiled again, until you close the process itself.

![image](https://github.com/AvaloniaUI/Avalonia/assets/79223742/204b0c8f-8784-4231-b420-bb2383b15732)

though any help in trying to figure out how to reference a name variable inside a DataTemplate where the buttons for that action also are would be greatly appreciated as well.

**Comment by SirZeeno** [comment]
Honestly im not sure either, i am pretty new to Avalonia and am mostly doing a lot of testing rn to see what works and what i need to do in certain situations like this where i need to reference a name in a DataTemplate. But my point on the issue is not what causes the crash, but the fact the process stays running after the crash and is only visible when going in the details tab in the task manager. it basically locks up the entire application from starting or being compiled again, until you close the process itself.

though any help in trying to figure out how to reference a name variable inside a DataTemplate where the buttons for that action also are would be greatly appreciated as well.

**Reply by timunie** [reply]
You shouldn't need to do that normally. Why do you need to access the buttons of a DataTemplate? I highly suggest to use Commands and Bindings to your ViewModel instead. 

Nonetheless, I think Rider is buggy in latest version. I see the same in latest EAP installed which was released recently iirc. But I don't think we can fix it on our side, so file an issue to them if non is on their tracker yet.
